File: lib/settings/data/models/app_settings_model.dart
import '../../domain/entities/app_settings.dart';

class AppSettingsModel extends AppSettingsEntity {
  const AppSettingsModel({
    required super.themeMode,
    required super.dbVersion,
    required super.updatedAt,
  });

  factory AppSettingsModel.fromEntity(AppSettingsEntity entity) {
    return AppSettingsModel(
      themeMode: entity.themeMode,
      dbVersion: entity.dbVersion,
      updatedAt: entity.updatedAt,
    );
  }

  AppSettingsEntity toEntity() {
    return AppSettingsEntity(
      themeMode: themeMode,
      dbVersion: dbVersion,
      updatedAt: updatedAt,
    );
  }
}

File: web/src/app/core/models/app-settings.model.ts
export interface AppSettings {
  key: string;
  value: string;
  createdAt?: string;
  updatedAt?: string;
}

export const DEFAULT_SETTINGS: AppSettings[] = [
  {
    key: 'theme_mode',
    value: 'system',
    createdAt: new Date().toISOString(),
    updatedAt: new Date().toISOString()
  }
];
